File: src/AddIns/Misc/PackageManagement/Project/Src/EnvDTE/ItemOperations.cs
// Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
// This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

using System;

namespace ICSharpCode.PackageManagement.EnvDTE
{
	public class ItemOperations : MarshalByRefObject, global::EnvDTE.ItemOperations
	{
		IPackageManagementFileService fileService;
		
		public ItemOperations(IPackageManagementFileService fileService)
		{
			this.fileService = fileService;
		}
		
		public void OpenFile(string fileName)
		{
			fileService.OpenFile(fileName);
		}
		
		public void Navigate(string url)
		{
			fileService.OpenFile(url);
		}
		
		public global::EnvDTE.Window NewFile(string fileName)
		{
			throw new NotImplementedException();
		}
	}
}

File: src/AddIns/Misc/PackageManagement/SharpDevelop.EnvDTE/Src/ItemOperations.vb
' Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
' This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

Namespace EnvDTE
	Public Interface ItemOperations
		Sub OpenFile(fileName As String)
		Sub Navigate(url As String)
		Function NewFile(fileName As String) As Window
	End Interface
End Namespace